AsyncDisplayKit/ASTableView.mm

```diff
@@ -455,6 +455,14 @@ - (void)reloadRowsAtIndexPaths:(NSArray *)indexPaths withRowAnimation:(UITableVi
   [_dataController reloadRowsAtIndexPaths:indexPaths withAnimationOptions:animation];
 }
 
+- (void)relayoutRowAtIndexPath:(NSIndexPath *)indexPath withRowAnimation:(UITableViewRowAnimation)animation
+{
+  ASDisplayNodeAssertMainThread();
+  ASCellNode *node = [self nodeForRowAtIndexPath:indexPath];
+  [node setNeedsLayout];
```

Comment:
-layoutIfNeeded is triggered by UITableView's -reloadRowsAtIndexPath:withRowAnimation (which is the next line). In the case of collection view, -layoutIfNeeded is triggered by -reloadItemsAtIndexPath. Do you think we still should call it anyway, or above assumptions are safe to make? Also, since -layoutIfNeeded is actually triggered with the current implementation, users can indeed capture the relayout's actions in an animation block and it should work.

This PR also adds animation to Kitten node's height change. I'm gonna push a commit that animates Kitten node's internal relayout as well.